src/components/layout/header.tsx

```diff
@@ -47,13 +52,51 @@ const Header = () => {
                 </div>
 
                 <div className="block md:hidden">
-                  <Button className="rounded bg-gray-100 p-2 text-gray-600 transition hover:text-gray-600/75">
+                  <Button
+                    className="rounded bg-gray-100 p-2 text-gray-600 transition hover:text-gray-600/75"
+                    onClick={toggleMobileMenu}
+                  >
                     <Image src={navMobileIcon} alt={"Mobile Nav Icon"} />
                   </Button>
                 </div>
               </div>
             </div>
           </div>
+          <div
+            className={clsx(
+              "fixed right-0 top-0 z-50 h-full w-72 transform flex-col items-center justify-center bg-white shadow-lg transition-transform duration-300 ease-in-out md:hidden",
+              {
+                "translate-x-full": !isMobileMenuOpen,
+                "translate-x-0": isMobileMenuOpen,
+              },
+            )}
+          >
+            <nav aria-label="Mobile Global" className="mt-10 p-2 text-sm">
+              <ul className="flex h-full w-full flex-col items-center justify-center gap-4 p-3">
+                {menus.map(({ name, href }) => (
+                  <li key={name} className="flex items-center p-1">
+                    <Link
+                      href={href}
+                      className={clsx({
+                        "text-gray-500 transition hover:text-gray-500/75": true,
+                        "border-b-2 border-gray-500/75": pathname === href,
+                      })}
+                      onClick={toggleMobileMenu}
+                    >
+                      {name}
+                    </Link>
+                  </li>
+                ))}
+              </ul>
+            </nav>
+          </div>
+          {isMobileMenuOpen && (
+            <div
+              className="fixed inset-0 z-40 bg-black opacity-25"
+              onClick={toggleMobileMenu}
+              aria-hidden="true"
+            />
+          )}
```

Comment:
Probably a good opportunity to break down into small components like MobileNav or something. Even better to do for the desktop one on top so that the file is not huge